prow/tide/tide.go

```diff
@@ -534,6 +533,37 @@ func subpoolsInParallel(goroutines int, sps map[string]*subpool, process func(*s
 	wg.Wait()
 }
 
+type queryData struct {
+	org         string
+	query       string
+	metricIndex int
```

Comment:
If you added a metric for the search query (or made the handling optional) you would not need to repeat yourself in this file.

Reply:
Actually this metric was added quite recently by google guys. They wanted to have a metric based on former outer loop index. So, it just about perservance of the things they wrote. Previously every goroutine started with this: ``org, q, i := org, q, i`` in the inner loop.

Reply:
Right - I'm pointing out that you add two sets of code here which are identical but for the handling of the metric. You could use the same code if:

- there's a metric in both cases
- metric changes are conditional on `i>0`

Reply:
Ok, so in package tide (which includes `status.go` and `tide.go`) I use one function ``queriesInParallel`` to handle case without metric and with metric, using one struct  `queryData`. Indeed, in  `status.go` index is useless, in `tide.go` I use it as you noticed.
Now you want me to:
1. Create another function for querying in pararell as using this struct in `status.go` is not quite right, since I do not use metricIndex there at all
2. Modify somehow the struct, do not create another function
3. Something else?

Sorry for being so descriptive, but I think I did not get the remark.

There is also similar function in `blockers.go` but it's another package and function itself is a little bit different. I just thaught, exporting it from package blockers to package tide is a little silly.

Reply:
I'm saying that this PR, in its current state, is checking in two large blocks of code that do an identical thing in an identical manner, aside from the metrics register. I would ask that the code be shared. I could see you sharing the code by either:

 - exposing errors in a metric for all types of queries (why should we see query errors in one place but not another?)
 - making the handling of the metric optional when none is specified in the `struct` you're operating over

Reply:
Putting aside metrics I think you actually referring to 2 similar functions in `blockers.go` and `tide.go`. Note that `blockers` is a separate package. I can make one function probably, but where to put it? Could you advise on that? There is no something like `utils.go `in tide. It more belongs to tide package but including it in tide package and exporting means importing the whole tide in blockers. I don't think it's desired.

Regarding metrics you are right, I can add them to the other queries as well. Wouldn't hurt.

Reply:
`prow/tide/queryutil` or similar should suffice and make it so that there are no import cycles. You may need to move the metrics around